exercises/fulfill_a_promise/problem.md

```diff
@@ -1,50 +1,41 @@
 # What is a promise?
-
-A promise is an object that defines a method called `then`.
-The promise object represents a value (or values) that may
-be available some time in the future.
-
-When a promise is resolved, any *"success functions"* registered
-with the `then` method will be called with the newly available
-data as arguments.
-
-If a promise is rejected then any *"failure functions"* registered
-with the `then` method will be called with the `Error` as argument.
-
-For now, do not be concerned with exactly **how** this works or what
-the syntax is.  We are about to dive into that in detail.
+A promise is an object that lets a consumer attach success and failure
+handlers to an asyncrhonous operation. If the operation succeeds,
+the function you register using the promises `then` function
+will be called with the result of the operation. If the operation
+fails, the function you register with the promises
+`catch` function is called.
+
+Just as the promise consumer can register for success or failure
+of the asynchronous operation, the producer of a promise needs to
+communicate success or failure. When the operation completes successfully,
+the producer calls a `resolve` function with the result of the operation.
+If the operation failed, the producer is responsible for calling the
+`reject` function with the error describing the failure.
```

Comment:
The wording here is not more beginner friendly, if that's the goal. It reads more dry & technical. "promise consumer can register for a success or failure" whaa?? how do I "register"? what does that mean? I recommend adding little inline examples:
```
aPromise.then(successHandler, errorHandler); //register a function to run on `resolve` and one for `reject`
```
+ one for resolve/rejecting. It's hard for beginners unfamiliar with this stuff to map "register" etc. to "pass a function to `.then`", a written example makes it clear.